File: A/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 2: 
            return [0, 1]
        num_map = {}
        for i in range(len(nums)): 
            difference = target - nums[i]
            if(difference in num_map): 
                return [i, num_map[difference]]
            num_map[nums[i]] = i
        return [0, 0]
        

File: A/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
  const numMap: Record<number, number> = {};

  for (let i = 0; i < nums.length; i++) {
    let difference = target - nums[i];
    if (difference in numMap) {
      return [i, numMap[difference]];
    }
    numMap[nums[i]] = i;
  }
  return [0, 0];
}
